qcodes/dataset/experiment_container.py

```diff
@@ -218,6 +219,10 @@ def new_experiment(name: str,
         the new experiment
     """
     conn = conn or connect(get_DB_location())
+    exp_ids = get_matching_exp_ids(conn, name=name, sample_name=sample_name)
```

Comment:
get_matching_exp_ids works in a different way. If you pass in sample_name=None it will take it to mean experiments that really have no sample name not any sample name

Reply:
Yes. I found out that is the problem
Trying to solve it

Reply:
Then, we need to bring back that function that I replaced

Reply:
Not necessarily. We could do something like ( I also think you should avoid renaming sample to sample_name)
```suggestion
   if sample is not None:
       arguments = {name: name, sample_name: sample}
   else:
       arguments = {name: name} 
    exp_ids = get_matching_exp_ids(conn, **arguments)
```

Reply:
Just discovered that we also have this method _create_exp_if_needed which is very similar https://github.com/QCoDeS/Qcodes/blob/master/qcodes/dataset/database_extract_runs.py#L122

Reply:
Thanks
Look into that